wallet-api/src/Wallet/API.hs

```diff
@@ -269,6 +283,34 @@ class WalletAPI m where
     -}
     slot :: m Slot
 
+-- | Change the 'validationInterval' of a 'WalletAPI' action.
+newtype WithValidationInterval m a = WithValidationInterval { runWithValidationInterval :: m (Slot, Slot) -> m a }
```

Comment:
> I think what you're suggesting is that we should set it "globally" for a WalletApi session

I was mainly looking for a way to make the default use case easy to write, but I suppose we can also add `createTxAndSubmit' = createTxAndSubmit defaultValidationTime`

> So I say just use function parameters rather than Reader.

Okay

Reply:
One approach that people take where you have a bunch of arguments with defaults for something is to have a `defaultOptions` record which you can pass, or modify it to set just the options you need. Not sure if that helps here.